components/Tabs.tsx

```diff
@@ -1,45 +1,56 @@
 'use client';
-import { cn } from '@/lib/utils';
-import { useFeedProvider } from '@/providers/FeedProvider';
-import { useNeynarProvider } from '@/providers/NeynarProvider';
-import { FeedType } from '@/types/Feed';
 import { Button } from '@/components/ui/button';
-import { useParams } from 'next/navigation';
+import useQueryParams from '@/hooks/useQueryParams';
+import { cn } from '@/lib/utils';
+import { useCallback, useEffect, useState } from 'react';
 
-type tab = {
-  label: string;
-  href?: string;
-  value: FeedType;
+interface Props {
+  tabs: {
+    label: string;
+    href?: string;
+    value: string;
+  }[];
+  className?: string;
+  onChange?: (value: string) => void;
 };
 
-export default function Tabs({ tabs, className = '' }: { tabs: tab[]; className?: string }) {
-  const { feedType, setFeedType } = useFeedProvider();
-  const { user } = useNeynarProvider();
-  const { username, channelId } = useParams();
+export default function Tabs({ tabs, className = '', onChange }: Props) {
+  const { setQueryParam, queryParams } = useQueryParams();
```

Comment:
can you move all `state` / `functions` 
- out of React component
- into a `hook` - ex. `useTabs`